azure/azure_infra/cpd-install.tf

```diff
@@ -582,3 +584,96 @@ resource "null_resource" "install_cpd_spss" {
         null_resource.install_cpd_ca
     ]
 }
+
+resource "null_resource" "install_cpd_watson_assistant" {
+    count = var.watson-assistant == "yes" && var.accept-cpd-license == "accept" ? 1 : 0
+    triggers = {
+        bootnode_ip_address = azurerm_public_ip.bootnode.ip_address
+        username = var.admin-username
+        private_key_file_path = var.ssh-private-key-file-path
+        namespace = var.cpd-namespace
+    }
+    connection {
+        type = "ssh"
+        host = azurerm_public_ip.bootnode.ip_address
+        user = var.admin-username
+        private_key = file(self.triggers.private_key_file_path)
+    }
+    provisioner "remote-exec" {
+        inline = [
+            "cat > ${local.installerhome}/watson-asst-override.yaml <<EOL\n${data.template_file.watson-asst-override.rendered}\nEOL",
+            "oc project ${self.triggers.namespace}",
+            "oc adm policy add-scc-to-group restricted system:serviceaccounts:${self.triggers.namespace}",
+            "docker_secret=$(oc get secrets | grep default-dockercfg | awk '{print $1}')",
+            "sed -i s/default-dockercfg-xxxxx/$docker_secret/g ${local.installerhome}/watson-asst-override.yaml",
+            "oc label --overwrite namespace ${self.triggers.namespace} ns=${self.triggers.namespace}",
+            "image_registry_route=$(oc get route -n  openshift-image-registry | grep image-registry | awk '{print $2}')",
+            "cpd-linux -r ${local.installerhome}/repo.yaml -a ibm-watson-assistant --version 1.4.2 -n ${self.triggers.namespace}  --transfer-image-to $image_registry_route/${self.triggers.namespace} --target-registry-username=kubeadmin --target-registry-password=$(oc whoami -t) --cluster-pull-prefix image-registry.openshift-image-registry.svc:5000/${self.triggers.namespace} --accept-all-licenses --override ${local.installerhome}/watson-asst-override.yaml --insecure-skip-tls-verify --verbose"
```

Comment:
Use service account token, like the other services do, rather than "oc whoami -t" (it expires)